client/my-sites/email/email-management/home/email-mailbox-action-menu.jsx

```diff
@@ -0,0 +1,260 @@
+ * @param {Function} gSuiteMenuParams.translate The translate function.
+ * @returns {Array|null} Returns an array of menu items or null if no items should be shown.
+ */
+const getGSuiteMenuItems = ( { account, email, mailbox, translate } ) => {
+	if ( hasGoogleAccountTOSWarning( account ) ) {
+		return null;
+	}
+
+	return [
+		{
+			href: getGmailUrl( email ),
+			image: gmailIcon,
+			imageAltText: translate( 'Gmail icon' ),
+			title: translate( 'View Gmail' ),
+		},
+		...( isEmailUserAdmin( mailbox )
+			? [
+					{
+						href: getGoogleAdminUrl( email ),
+						image: googleAdminIcon,
+						imageAltText: translate( 'Google Admin icon' ),
+						title: translate( 'View Admin' ),
+					},
+			  ]
+			: [] ),
+		{
+			href: getGoogleCalendarUrl( email ),
+			image: googleCalendarIcon,
+			imageAltText: translate( 'Google Calendar icon' ),
+			title: translate( 'View Calendar' ),
+		},
+		{
+			href: getGoogleDocsUrl( email ),
+			image: googleDocsIcon,
+			imageAltText: translate( 'Google Docs icon' ),
+			title: translate( 'View Docs' ),
+		},
+		{
+			href: getGoogleDriveUrl( email ),
+			image: googleDriveIcon,
+			imageAltText: translate( 'Google Drive icon' ),
+			title: translate( 'View Drive' ),
+		},
+		{
+			href: getGoogleSheetsUrl( email ),
+			image: googleSheetsIcon,
+			imageAltText: translate( 'Google Sheets icon' ),
+			title: translate( 'View Sheets' ),
+		},
+		{
+			href: getGoogleSlidesUrl( email ),
+			image: googleSlidesIcon,
+			imageAltText: translate( 'Google Slides icon' ),
+			title: translate( 'View Slides' ),
+		},
+	];
+};
+
+const getEmailForwardMenuItems = ( {
+	currentRoute,
+	dispatch,
+	domain,
+	mailbox,
+	selectedSite,
+	translate,
+} ) => {
+	return [
+		{
+			href: emailManagementForwarding( selectedSite.slug, domain.name, currentRoute ),
+			isInternalLink: true,
+			materialIcon: 'create',
+			title: translate( 'Edit', {
+				comment: 'Edit an email forward',
+			} ),
+		},
+		{
+			isInternalLink: true,
+			materialIcon: 'delete',
+			onClick: () => {
+				removeEmailForwardMailbox( { dispatch, mailbox } );
+			},
+			key: `remove_forward:${ mailbox.mailbox }`,
+			title: translate( 'Remove email forward', {
+				comment: 'Remove an email forward',
+			} ),
+		},
+	];
+};
+
+const EmailMailboxActionMenu = ( { account, domain, mailbox, selectedSite } ) => {
+	const currentRoute = useSelector( getCurrentRoute );
+	const dispatch = useDispatch();
+	const translate = useTranslate();
+
+	const email = `${ mailbox.mailbox }@${ mailbox.domain }`;
+
+	const getMenuItems = () => {
+		if ( hasTitanMailWithUs( domain ) ) {
+			return getTitanMenuItems( { email, translate } );
+		}
+
+		if ( hasGSuiteWithUs( domain ) ) {
+			return getGSuiteMenuItems( { account, email, mailbox, translate } );
+		}
+
+		if ( hasEmailForwards( domain ) ) {
+			return getEmailForwardMenuItems( {
+				currentRoute,
+				dispatch,
+				domain,
+				mailbox,
+				selectedSite,
+				translate,
+			} );
+		}
+
+		return null;
+	};
```

Comment:
Not mandatory but it looks like this `getMenuItems()` function could leave outside of `EmailMailboxActionMenu`.

Reply:
It was originally there, but it then meant that we were essentially copying all the arguments in both functions, which were both pretty much pass-throughs. So I shifted this inside the container function so we can remove the really tall function signature from the helper function.